cpp/src/arrow/flight/sql/server_test.cc

```diff
@@ -381,6 +382,55 @@ TEST_F(TestFlightSqlServer, TestCommandGetTablesWithIncludedSchemas) {
   AssertTablesEqual(*expected_table, *table);
 }
 
+TEST_F(TestFlightSqlServer, TestCommandGetTypeInfo) {
+  ASSERT_OK_AND_ASSIGN(auto flight_info, sql_client->GetTypeInfo({}));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  const std::shared_ptr<Schema>& expected_schema = SqlSchema::GetTypeInfoSchema();
+
+  const std::shared_ptr<RecordBatch>& batch =
+      example::DoGetTypeInfoResult(expected_schema);
+
+  const arrow::Result<std::shared_ptr<Table>>& expected_table_result =
+      Table::FromRecordBatches({batch});
+
+  std::shared_ptr<Table> table;
+  std::shared_ptr<Table> expected_table;
+  ASSERT_OK(stream->ReadAll(&table));
+
+  ASSERT_OK_AND_ASSIGN(expected_table, expected_table_result);
+
+  ASSERT_TRUE(table->schema()->Equals(*expected_schema));
```

Comment:
`AssertTablesEqual` already checks the schema.

Reply:
Fixed